apm-agent-plugins/apm-jaxrs-plugin/src/test/java/co/elastic/apm/agent/jaxrs/JaxRsTransactionNameInstrumentationTest.java

```diff
@@ -128,25 +131,80 @@ public void testProxyClassInstrumentationExclusion() {
     @Test
     public void testJaxRsTransactionNameNonSampledTransactions() throws IOException {
         config.getConfig(CoreConfiguration.class).getSampleRate().update(0.0, SpyConfiguration.CONFIG_SOURCE_NAME);
-
         ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install());
 
         doRequest("test");
+
         List<Transaction> actualTransactions = reporter.getTransactions();
         assertThat(actualTransactions).hasSize(1);
         assertThat(actualTransactions.get(0).getName().toString()).isEqualTo("ResourceWithPath#testMethod");
     }
 
+    @Test
+    public void testJaxRsTransactionNameFromPathAnnotationInheritanceEnabled() {
+        when(config.getConfig(CoreConfiguration.class).isUseAnnotationValueForTransactionName()).thenReturn(true);
+        when(config.getConfig(JaxRsConfiguration.class).isEnableJaxrsAnnotationInheritance()).thenReturn(true);
+
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install());
+
+        doRequest("test");
+        doRequest("testAbstract");
+        doRequest("testInterface");
+
+        List<Transaction> actualTransactions = reporter.getTransactions();
+        assertThat(actualTransactions).hasSize(3);
+        assertThat(actualTransactions.get(0).getName().toString()).isEqualTo("GET /test");
+        assertThat(actualTransactions.get(1).getName().toString()).isEqualTo("GET /testAbstract");
+        assertThat(actualTransactions.get(2).getName().toString()).isEqualTo("GET /testInterface");
+    }
+
+    @Test
+    public void testJaxRsTransactionNameFromPathAnnotationInheritanceDisabled() {
+        when(config.getConfig(CoreConfiguration.class).isUseAnnotationValueForTransactionName()).thenReturn(true);
+        when(config.getConfig(JaxRsConfiguration.class).isEnableJaxrsAnnotationInheritance()).thenReturn(false);
+
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install());
+
+        doRequest("test");
+        doRequest("testInterface");
+        doRequest("testAbstract");
+
+        List<Transaction> actualTransactions = reporter.getTransactions();
+        assertThat(actualTransactions).hasSize(3);
+        assertThat(actualTransactions.get(0).getName().toString()).isEqualTo("GET /test");
+        assertThat(actualTransactions.get(1).getName().toString()).isEqualTo("unnamed");
+        assertThat(actualTransactions.get(2).getName().toString()).isEqualTo("unnamed");
```

Comment:
```suggestion
        assertThat(actualTransactions.get(2).getName().toString()).isEqualTo("GET /testAbstract");
```

Reply:
@felixbarny  in current test case with:
 ```
when(config.getConfig(CoreConfiguration.class).isUseAnnotationValueForTransactionName()).thenReturn(true);
when(config.getConfig(JaxRsConfiguration.class).isEnableJaxrsAnnotationInheritance()).thenReturn(false);
```
Should we capture annotation value for transaction name?

Reply:
Ah, I missed that `isEnableJaxrsAnnotationInheritance()).thenReturn(false);`. `unnamed` looks correct then